app/scripts/controllers/createFromURL.js

```diff
@@ -0,0 +1,183 @@
+      $scope.alerts[alert.name] = alert.data;
+    });
+    AlertMessageService.clearAlerts();
+
+    $scope.alerts = {};
+    $scope.selected = {};
+
+    var alertInvalidApp = function(app) {
+      $scope.alerts.invalidImageStream = {
+        type: "error",
+        message: "The app name \"" + app + "\" is not valid.  An app name is an alphanumeric (a-z, and 0-9) string with a maximum length of 24 characters, where the first character is a letter (a-z), and the '-' character is allowed anywhere except the first or last character."
+      };
+    };
+
+    var alertInvalidImageStream = function(imageStream) {
+      $scope.alerts.invalidImageStream = {
+        type: "error",
+        message: "The requested image stream \"" + imageStream + "\" could not be loaded."
+      };
+    };
+
+    var alertInvalidImageTag = function(imageTag) {
+      $scope.alerts.invalidImageTag = {
+        type: "error",
+        message: "The requested image stream tag \"" + imageTag + "\" could not be loaded."
+      };
+    };
+
+    var alertInvalidNamespace = function(namespace) {
+      $scope.alerts.invalidNamespace = {
+        type: "error",
+        message: "Resources from the namespace \"" + namespace + "\" are not permitted."
+      };
+    };
+
+    var alertInvalidTemplate = function(template) {
+      $scope.alerts.invalidTemplate = {
+        type: "error",
+        message: "The requested template \"" + template + "\" could not be loaded."
+      };
+    };
+
+    var alertResourceRequired = function() {
+      $scope.alerts.resourceRequired = {
+        type: "error",
+        message: "An image stream or template is required."
+      };
+    };
+
+    var showInvalidResource = function() {
+      $scope.alerts.invalidResource = {
+        type: "error",
+        message: "Image streams and templates cannot be combined."
+      };
+    };
+
+    var getTemplateParamsMap = function () {
+      try {
+        return $routeParams.templateParamsMap && JSON.parse($routeParams.templateParamsMap) || {};
+      }
+      catch (e) {
+        $scope.alerts.invalidTemplateParams = {
+          type: "error",
+          message: "The templateParamsMap is not valid JSON. " + e
+        };
+      }
+    };
+
+    var namespaceWhitelist = window.OPENSHIFT_CONSTANTS.CREATE_FROM_URL_WHITELIST;
+    var whiteListedCreateDetailsKeys = ['namespace', 'app', 'imageStream', 'imageTag', 'sourceURI', 'sourceRef', 'contextDir', 'template', 'templateParamsMap'];
+    var createDetails = _.pick($routeParams, function(value, key) {
+      // routeParams without a value (e.g., ?app&) return true, which results in "true" displaying in the UI
+      return _.contains(whiteListedCreateDetailsKeys, key) && _.isString(value);
+    });
+    // if no namespace is specified, set it to 'openshift'
+    createDetails.namespace = createDetails.namespace || 'openshift';
+
+    var validateApp = function (app) {
+      return /^[a-z]([-a-z0-9]*[a-z0-9])?$/.test(app);
```

Comment:
I'm thinking we shouldn't change this to `name` so it's not confused with the `app` label. It's also consistent with `oc new-app` and the label we uses in the web console.

```
--name='': Set name to use for generated application artifacts
```